src/components/measureLanding/measureList/measureSearch/search.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { Dispatch, SetStateAction } from "react";
+import { IconButton, MenuItem } from "@mui/material";
+import _ from "lodash";
+import ClearIcon from "@mui/icons-material/Clear";
+import SearchIcon from "@mui/icons-material/Search";
+import { useFormik } from "formik";
+import "twin.macro";
+import "styled-components/macro";
+import { Select, TextField } from "@madie/madie-design-system/dist/react";
+import { MeasureSearchCriteria } from "../../MeasureLanding";
+
+const renderMenuItemsForFilter = (options: string[]) => {
+  return [
+    ...options.map((option) => (
+      <MenuItem
+        key={`${option}-option`}
+        value={option}
+        data-testid={`${option}-option`}
+      >
+        {option}
+      </MenuItem>
+    )),
+  ];
+};
+
+const Search = (props: {
+  searchCriteria: MeasureSearchCriteria;
+  setSearchCriteria: Dispatch<SetStateAction<MeasureSearchCriteria>>;
+  handlePageChange: (e, v) => void;
+}) => {
+  const { searchCriteria, setSearchCriteria, handlePageChange } = { ...props };
+  const formik = useFormik({
+    initialValues: {
+      searchField: searchCriteria?.searchField,
+      filterBy: searchCriteria?.optionalSearchProperties?.[0],
+    },
+    enableReinitialize: true,
+    onSubmit: async (values) => {
+      setSearchCriteria({
+        searchField: values?.searchField,
+        optionalSearchProperties: [values?.filterBy],
+      });
+      handlePageChange(null, 1);
+    },
+  });
+
+  return (
+    <form
+      onSubmit={formik.handleSubmit}
+      tw="col-span-3 grid grid-cols-3 gap-4 items-end"
+    >
+      <Select
+        defaultValue=""
+        placeHolder={{ name: "Filter By", value: "" }}
+        label="Filter By"
+        name="filterBy"
+        id={`filter-by`}
+        data-testid={`filter-by`}
+        inputProps={{
+          "data-testid": `filter-by-input`,
+        }}
+        {...formik.getFieldProps("filterBy")}
+        options={renderMenuItemsForFilter([
+          "-",
+          "Measure",
+          "Version",
+          "Model",
+          "CMS ID",
```

Comment:
It might be good to index these columns in the database sometime in the future, but for now it looks like its performing really well. Good job!

Reply:
Yup make sense to index these fields, Can be reviewed if the PROD has performance issues